sources/npmRegistryUtils.ts

```diff
@@ -32,30 +34,58 @@ export async function fetchAsJson(packageName: string, version?: string) {
   return httpUtils.fetchAsJson(`${npmRegistryUrl}/${packageName}${version ? `/${version}` : ``}`, {headers});
 }
 
-export function verifySignature({signatures, integrity, packageName, version}: {
+const getVerificationKeys = once(async (): Promise<Array<{ keyid: string, key: KeyObject }>> => {
+  let keys: Array<{ keyid: string, key: string }>;
+  if (process.env.COREPACK_INTEGRITY_KEYS) {
+    // We use the format of the `GET /-/npm/v1/keys` endpoint with `npm` instead
+    // of `keys` as the wrapping key.
+    const keysFromEnv = JSON.parse(process.env.COREPACK_INTEGRITY_KEYS) as { npm: Array<{ keyid: string, key: string }> };
+    keys = keysFromEnv.npm.map(k => ({keyid: k.keyid, key: k.key}));
+    debugUtils.log(`Using COREPACK_INTEGRITY_KEYS to verify signatures: ${keys.map(k => k.keyid).join(`, `)}`);
+  } else {
+    // This follows the implementation for npm.
+    // See https://github.com/npm/cli/blob/3a80a7b7d168c23b5e297cba7b47ba5b9875934d/lib/utils/verify-signatures.js#L174
+    // We only support sigstore for NPM. For other registries
+    // COREPACK_INTEGRITY_KEYS can be used.
+    const sigstoreTufClient = await sigstoreTuf.initTUF();
```

Comment:
We probably want to provide a cachePath, as npm is doing https://github.com/npm/cli/blob/3a80a7b7d168c23b5e297cba7b47ba5b9875934d/workspaces/config/lib/definitions/definitions.js#L315
```suggestion
    const sigstoreTufClient = await sigstoreTuf.initTUF({ cachePath: path.join(folderUtils.getCorepackHomeFolder(), `_tuf`) });
```